actioncable/app/javascript/action_cable/index.js

```diff
@@ -30,14 +30,20 @@ export function getConfig(name) {
 }
 
 export function createWebSocketURL(url) {
-  if (url && !/^wss?:/i.test(url)) {
+  let webSocketURL
+  if (typeof url === 'function') {
+    webSocketURL = url()
+  } else {
+    webSocketURL = url
+  }
```

Comment:
I'm no JS-hound, but is it possible to use a ternary here? Ala `const webSocketURL = (typeof url === 'function') ? url() : url`

We should add a newline to separate the assignment and the if below regardless though.

Reply:
yeah it can definitely be done with a ternary, sometimes people think they are less readable so I didn't do it here, but easy enough to switch. I'll add a space as well.